src/tests/roc_audio/test_watchdog.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2017 Mikhail Baranov
+ * Copyright (c) 2017 Victor Gaydov
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_audio/watchdog.h"
+#include "roc_core/buffer_pool.h"
+#include "roc_core/heap_allocator.h"
+#include "roc_core/slice.h"
+
+namespace roc {
+namespace audio {
+
+namespace {
+
+enum {
+    MaxBufSize = 5000,
+
+    SamplesPerFrame = 200,
+    NumCh = 2,
+
+    Timeout = 20
```

Comment:
```
    SamplesPerFrame = 200,
    Timeout = 20
```

I don't think it makes any sense to test the case when frame is larger than timeout.